src/runtime-rs/crates/runtimes/virt_container/src/sandbox.rs

```diff
@@ -188,15 +196,33 @@ impl Sandbox for VirtSandbox {
         info!(sl!(), "start vm");
 
         // execute pre-start hook functions, including Prestart Hooks and CreateRuntime Hooks
-        let (prestart_hooks, create_runtime_hooks) = match spec.hooks.as_ref() {
-            Some(hooks) => (hooks.prestart.clone(), hooks.create_runtime.clone()),
-            None => (Vec::new(), Vec::new()),
+        let (prestart_hooks, create_runtime_hooks, has_prestart_hooks) = match spec.hooks.as_ref() {
```

Comment:
I am not sure the parameter has_prestart_hooks is correct here. Sometimes the hooks may only contain start_container hooks, for example, which is not prestart_hook. And I following operation seems to be strongly related to create_runtime hooks.

Reply:
fixed